source/SpinalHDL/Data types/Vec.rst

```diff
@@ -85,6 +85,8 @@ Comparison
    val vec1 = Vec(SInt(8 bits), 2)
 
    myBool := vec2 === vec1  // Compare all elements
+   // is equivalent to:
```

Comment:
Yes I find it possibly confusing because the equivalence is only for the specific case of Vecs with 2 elements.
Maybe we make it an "in this case equivalent..."?
No need to uncomment it, if you want it to be compilable we could just make it a `val otherBool = ...` - but in the end many of these examples show single lines of usage and would not be very useful taken together... (which was one of the things that stopped me when I tried to move them to the compiled project - much to rewrite and IMO little gain in trying to find the n-th way of saying "someBool"...)

Reply:
I don't understand how it changes if there are 3 element or 100 elements.

To me it would be a "given" (that "in this case equivalent...").  I would expect the equivalent to relative directly to what is near-by and only for this example.  If I do something different may no longer hold true.

So the question is how much change is needed to get an unexpected result ?  That is not clear to me, can you think of an example ?

From the equivalent example the reader should be able to infer how more equality `===` operators are used and ANDed to reduce to a single Bool result.  Also how more element are compared for equality purposes.

If it helps we can use an additional 3 element example instead for clarify this point:

```
result := vec(0).a === vec(1).a && vec(1).a === vec(2).a && vec(2).a === vec(0).a &&
          vec(0).b === vec(1).b && vec(1).b === vec(2).b && vec(2).b === vec(0).b &&
          vec(0).c === vec(1).c && vec(1).c === vec(2).c && vec(2).c === vec(0).c
// although we can expect boolean algebra reduction to remove some of those comparisons when possible
//  AA === BB && BB === CC therefore CC === AA eliminating the need for that last comparison
```

This would be my first guess at how that works in a difference scenario.  Maybe SpinalHDL performs the boolean algebra reduction, maybe my hardware synthesis flow stage does, does not matter, unless we are trying to relate the example more to specifically how HDL will be generated ?  Functional equivalence (instead of absolute detail) is good enough for me in such an example.

So the best guess by the reader about a different scenario (than the example) is correct.

So how do you get an unexpected result that bites you in the ass ?

Reply:
Rereading it I agree, it should be clear from the context.